mongodbatlas/resource_mongodbatlas_online_archive_test.go

```diff
@@ -331,6 +378,59 @@ func testAccBackupRSOnlineArchiveConfigWithoutSchedule(orgID, projectName, clust
 	`, testAccBackupRSOnlineArchiveConfigFirstStep(orgID, projectName, clusterName))
 }
 
+func testAccBackupRSOnlineArchiveConfigWithProcessRegion(orgID, projectName, clusterName, cloudProvider, region string) string {
+	return fmt.Sprintf(`
+	%s
+	resource "mongodbatlas_online_archive" "users_archive" {
+		project_id = mongodbatlas_cluster.online_archive_test.project_id
+		cluster_name = mongodbatlas_cluster.online_archive_test.name
+		coll_name = "listingsAndReviews"
+		collection_type = "STANDARD"
+		db_name = "sample_airbnb"
+
+		criteria {
+			type = "DATE"
+			date_field = "last_review"
+			date_format = "ISODATE"
+			expire_after_days = 2
+		}
+
+		partition_fields {
+			field_name = "last_review"
+			order = 0
+		}
+
+		partition_fields {
+			field_name = "maximum_nights"
+			order = 1
+		}
+
+		partition_fields {
+			field_name = "name"
+			order = 2
+		}
+
+		data_process_region {
+			cloud_provider = %[2]q
+			region = %[3]q
+		}
+
+		sync_creation = true
+	}
+
+	data "mongodbatlas_online_archive" "read_archive" {
+		project_id =  mongodbatlas_online_archive.users_archive.project_id
+		cluster_name = mongodbatlas_online_archive.users_archive.cluster_name
+		archive_id = mongodbatlas_online_archive.users_archive.archive_id
+	}
+
+	data "mongodbatlas_online_archives" "all" {
```

Comment:
@marcosuma @AgustinBettati this is an example where we can have flaky tests because using the plural data source, it doesn't necessarily run after the resources are created as there's no dependency with them and can be executed at any time

Reply:
Are you sure? I remember that as long as you specify a variable instead of a value it'll wait until the resource to which the variable refers to is ready.

Reply:
I have the same idea as Marco, as long as you reference an attribute from the resource it will wait for its completion.

Reply:
i might be mistaken but this is what I concluded while working in search-index.
If the values are not computed there is not a real dependency.

Reply:
Gonna resolve this because in any case this is not a new thing being introduced in this PR. If we realise it's like @lantoli is saying we can update it in all the places altogether.